frontend/src/components/shared/curies/CurieConceptSelector.jsx

```diff
@@ -0,0 +1,189 @@
+import React, { useRef, useEffect } from 'react';
+
+import ReactDOM from 'react-dom';
+
+import {
+  FormControl, Button, Badge, InputGroup,
+} from 'react-bootstrap';
+import shortid from 'shortid';
+
+import { AutoSizer, List } from 'react-virtualized';
+
+import entityNameDisplay from '@/utils/entityNameDisplay';
+import getNodeTypeColorMap from '@/utils/colorUtils';
+import curieUrls from '@/utils/curieUrls';
+
+export default function CurieConceptSelector({
+  concepts,
+  curies,
+  selection, handleSelect,
+  searchTerm, updateSearchTerm,
+  loading,
+  rightButtonFunction, rightButtonContents,
+  focus, clearFocus,
+}) {
+  // Reference to the input element used for setting focus
+  const inputRef = useRef(null);
+  useEffect(() => {
+    if (!focus) return;
+    // We have to use the deprecated find-dom-node method
+    // because react-bootstrap doesn't expose an alternative interface
+
+    // eslint-disable-next-line react/no-find-dom-node
+    const node = ReactDOM.findDOMNode(inputRef.current);
```

Comment:
The property on the FormControl should be inputRef instead of just ref and will set the correct ref so you can just call inputRef.current.focus().